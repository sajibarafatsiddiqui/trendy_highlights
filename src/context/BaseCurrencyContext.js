import React from 'react';

const BaseCurrencyContext = React.createContext();
export default BaseCurrencyContext;
