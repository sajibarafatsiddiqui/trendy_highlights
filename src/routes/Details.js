import React from 'react';

const Details = () => (
  <div><h1>hello</h1></div>
);

export default Details;
