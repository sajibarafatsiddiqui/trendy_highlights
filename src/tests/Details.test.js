import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import axios from 'axios';
import Details from 'routes/Details';
import { Provider } from 'react-redux';
import BaseCurrencyContext from 'context/BaseCurrencyContext';
import {
  MemoryRouter, Route, Routes,
} from 'react-router-dom';
import store from 'redux/store';

jest.mock('axios');
axios.get.mockResolvedValue({
  data: {
    response: {
      rates: {
        AED: 3.6725,
      },
    },
  },
});

const baseCurrency = 'USD';

const wrap = (component) => (
  <Provider store={store}>
    <BaseCurrencyContext.Provider value={{ baseCurrency }}>
      <MemoryRouter initialEntries={['/details/USD-AED']}>
        <Routes>
          <Route path="/details/:currencies" element={component} />
        </Routes>
      </MemoryRouter>
    </BaseCurrencyContext.Provider>
  </Provider>
);

describe('Tests for <Details /> component', () => {
  render(wrap(<Details />));
  it('Should render childs from API correctly', () => {
    expect(screen.getByText('AED')).not.toBeNull();
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(wrap(<Details />)).toJSON()).toMatchSnapshot();
  });
});
