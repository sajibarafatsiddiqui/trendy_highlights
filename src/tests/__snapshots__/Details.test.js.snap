// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Home /> component Should match the snapshot 1`] = `
Array [
  <p
    className="dark_header"
  >
    Exchange Rate
  </p>,
  <div
    className="g-0 row row-cols-md-1 row-cols-1"
  >
    <div
      className="col"
    >
      <div
        className="USD BaseCard card"
      >
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          />
          <p
            className="card-text"
          />
          <div
            className="card-footer"
          >
            <h1>
              USD
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="g-0 row row-cols-md-1 row-cols-1"
  >
    <p
      className="dark card-text"
    >
      Exchange Rate By Currency
    </p>
  </div>,
  <div
    className="g-0 row row-cols-md-2 row-cols-2"
  />,
]
`;
